brainrender_napari/widgets/atlas_manager_view.py

```diff
@@ -19,16 +19,48 @@
 from brainglobe_atlasapi.update_atlases import install_atlas, update_atlas
 from napari.qt import thread_worker
 from qtpy.QtCore import Signal
-from qtpy.QtWidgets import QTableView, QWidget
+from qtpy.QtWidgets import (
+    QProgressBar,
+    QTableView,
+    QWidget,
+)
 
 from brainrender_napari.data_models.atlas_table_model import AtlasTableModel
 from brainrender_napari.utils.formatting import format_atlas_name
 from brainrender_napari.widgets.atlas_manager_dialog import AtlasManagerDialog
 
 
+def _format_bytes(num_bytes: float) -> str:
+    for unit in ["B", "KB", "MB", "GB"]:
+        if num_bytes < 1024:
+            return f"{num_bytes:.2f} {unit}"
+        num_bytes /= 1024
+    return f"{num_bytes:.2f} TB"
+
+
+def install_atlas_with_progress(atlas_name: str, fn_update: Callable):
+    """
+    By passing fn_update when instantiating BrainGlobeAtlas
+    Progress information during the downlaod progress is
+    retrieved so that the actual progress (%) can be updated
+    """
+    install_atlas(atlas_name, fn_update=fn_update)
+    return atlas_name
+
+
+def update_atlas_with_progress(atlas_name: str, fn_update: Callable):
+    """
+    When updating an existing atlas, it should also be possible to pass
+    fn_update to get the actual progress (%)
+    """
+    update_atlas(atlas_name, fn_update=fn_update)
+    return atlas_name
```

Comment:
It feels like this functions don't add much, could we just call the `brainglobe_atlasapi` functions directly where we need these? But maybe I am missing something?